components/match2/wikis/fifa/match_group_input_custom.lua

```diff
@@ -0,0 +1,624 @@
+		return true
+	end
+
+	-- Match is automatically marked finished upon page edit after a
+	-- certain amount of time (depending on whether the date is exact)
+	if obj.timestamp and obj.timestamp > DateExt.epochZero then
+		local threshold = obj.dateexact and 30800 or 86400
+		if obj.timestamp + threshold < NOW then
+			return true
+		end
+	end
+end
+
+function CustomMatchGroupInput._determineWinnerIfMissing(scores)
+	local maxScore = math.max(unpack(scores or {0}))
+	-- if we have a positive score and the match is finished we also have a winner
+	if maxScore > 0 then
+		if Array.all(scores, function(score) return score == maxScore end) then
+			return 0
+		end
+
+		for opponentIndex, score in pairs(scores) do
+			if score == maxScore then
+				return opponentIndex
+			end
+		end
+	end
+end
+
+function CustomMatchGroupInput._setPlacements(match)
+	for opponentIndex = 1, MAX_NUM_OPPONENTS do
+		local opponent = match['opponent' .. opponentIndex]
+
+		if match.winner == opponentIndex or match.winner == 0 then
+			opponent.placement = 1
+		elseif match.winner then
+			opponent.placement = 2
+		end
+	end
+end
+
+function CustomMatchGroupInput._opponentInput(match)
+	local opponentIndex = 1
+	local opponent = match['opponent' .. opponentIndex]
+
+	while opponentIndex <= MAX_NUM_OPPONENTS and Logic.isNotEmpty(opponent) do
+		opponent = Json.parseIfString(opponent) or Opponent.blank()
+
+		-- Convert byes to literals
+		if
+			string.upper(opponent.template or '') == BYE
+			or string.upper(opponent.name or '') == BYE
+		then
+			opponent = {type = Opponent.literal, name = BYE}
+		end
+
+		--process input
+		if opponent.type == Opponent.team or opponent.type == Opponent.solo or
+			opponent.type == Opponent.literal then
+
+			opponent = CustomMatchGroupInput.processOpponent(opponent, match.timestamp)
+		else
+			error('Unsupported Opponent Type')
+		end
+
+		match['opponent' .. opponentIndex] = opponent
+
+		opponentIndex = opponentIndex + 1
+		opponent = match['opponent' .. opponentIndex]
+	end
+end
+
+function CustomMatchGroupInput.processOpponent(record, timestamp)
+	local opponent = Opponent.readOpponentArgs(record)
+		or Opponent.blank()
+
+	local teamTemplateDate = timestamp
+	-- If date is epoch, resolve using tournament dates instead
+	-- Epoch indicates that the match is missing a date
+	-- In order to get correct child team template, we will use an approximately date and not 1970-01-01
+	if teamTemplateDate == DateExt.epochZero then
+		teamTemplateDate = Variables.varDefaultMulti(
+			'tournament_enddate',
+			'tournament_startdate',
+			NOW
+		)
+	end
+
+	Opponent.resolve(opponent, teamTemplateDate)
+
+	MatchGroupInput.mergeRecordWithOpponent(record, opponent)
+
+	if type(record.players) == 'table' then
+		for _, player in pairs(record.players) do
+			player.name = player.name:gsub(' ', '_')
+		end
+	end
+
+	if record.name and record.type ~= Opponent.literal then
+		record.name = record.name:gsub(' ', '_')
+	end
```

Comment:
Why this?

Reply:
(I removed this in the newest revision now) I cant remember it but I think it was during the early development where I run into issue with GTL (the **/next/downstream** one) skipping names with spaces. 

I've removed this in dev and the issue seems to not exist anmore so I guess its leftover (If I recall, the gsub thing were solved through Legacy)